tests/types/test_mxc_uri.py

```diff
@@ -0,0 +1,94 @@
+# Copyright 2022 The Matrix.org Foundation C.I.C.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from unittest import TestCase
+
+from matrix_common.types.mxc_uri import MXCUri
+
+
+class MXCUriTestCase(TestCase):
+    def test_valid_mxc_uris(self) -> None:
```

Comment:
This seems to be two test-cases combined:

Valid MXCs, both parsing and dumping to string. It might be better to separate those.